### PR TITLE
Fix labels in events being treated as full label objects

### PR DESCRIPTION
**Pull request by joseartrivera**
#### Description
The Get-GitHubEvent is currently broken if the issue contains and event that includes the "label" or "labels" property. This is because Add-GitHubEventAdditionalProperties treats the labels in the event object as full fledged label objects that normally contain a URL property and other additional pieces of data. The label property in events is a lighter-weight object that simply contains the string for the label and color that was changed, it is not the full object that you would typically receive from a call like Get-GithubLabel.

#### Issues Fixed
none

#### References
The docs do not explicitly call this out, but you can check the payload in fiddler for verificaiton.

#### Checklist

- [x] You actually ran the code that you just wrote, especially if you did just "one last quick change".
- [ ] Comment-based help added/updated, including examples.
- [ ] [Static analysis](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#static-analysis) is reporting back clean.
- [x] New/changed code adheres to our [coding guidelines](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#coding-guidelines).
- [ ] [Formatters were created](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#formatters) for any new types being added.
- [x] New/changed code continues to [support the pipeline](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#pipeline-support).
- [ ] Changes to the manifest file follow the [manifest guidance](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#module-manifest).
- [ ] Unit tests were added/updated and are all passing. See [testing guidelines](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#testing).  This includes making sure that all pipeline input variations have been covered.
- [ ] Relevant usage examples have been added/updated in [USAGE.md](https://github.com/microsoft/PowerShellForGitHub/blob/master/USAGE.md).
- [ ] If desired, ensure your name is added to our [Contributors list](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#contributors)

### CODE REVIEW

**Review by HowardWolosky — CHANGES_REQUESTED** [review]
Thanks for the report.  I was able to repro your issue by calling 

```powershell
Get-GitHubEvent -OwnerName microsoft -RepositoryName PowerShellForGitHub -Issue 306
```

What you supplied would indeed fix the specific issue you're running into, but I don't know if it's actually the _right_ fix.

I think the more complete fix would be to create a new `Type` called `GitHub.LabelSummary` for the reduce content, similar to what we do for [`GitHub.Team`/`GitHub.TeamSummary`](https://github.com/microsoft/PowerShellForGitHub/blob/d4997057f8b1234ea1aabeb4fb6742148d3afaaf/GitHubTeams.ps1#L5-L6), and then update `Add-GitHubLabelAdditionalProperties` to optionally take in the `OwnerName` and `RepositoryName`.  If supplied, and `$item.url` is not defined, then those should be used for forming the `$repositoryUrl`.  Then you'd update [`Add-GitHubEventAdditionalProperties`](https://github.com/microsoft/PowerShellForGitHub/blob/d4997057f8b1234ea1aabeb4fb6742148d3afaaf/GitHubEvents.ps1#L238-L246) to pass in the `ReposistoryUrl` and `OwnerName` along with the new `TypeName`.

That should fix the bug while also ensuring individual labels can still be passed via the pipeline into other functions.

**Comment by joseartrivera** [comment]
> Thanks for the report. I was able to repro your issue by calling
> 
> ```powershell
> Get-GitHubEvent -OwnerName microsoft -RepositoryName PowerShellForGitHub -Issue 306
> ```
> 
> What you supplied would indeed fix the specific issue you're running into, but I don't know if it's actually the _right_ fix.
> 
> I think the more complete fix would be to create a new `Type` called `GitHub.LabelSummary` for the reduce content, similar to what we do for [`GitHub.Team`/`GitHub.TeamSummary`](https://github.com/microsoft/PowerShellForGitHub/blob/d4997057f8b1234ea1aabeb4fb6742148d3afaaf/GitHubTeams.ps1#L5-L6), and then update `Add-GitHubLabelAdditionalProperties` to optionally take in the `OwnerName` and `RepositoryName`. If supplied, and `$item.url` is not defined, then those should be used for forming the `$repositoryUrl`. Then you'd update [`Add-GitHubEventAdditionalProperties`](https://github.com/microsoft/PowerShellForGitHub/blob/d4997057f8b1234ea1aabeb4fb6742148d3afaaf/GitHubEvents.ps1#L238-L246) to pass in the `ReposistoryUrl` and `OwnerName` along with the new `TypeName`.
> 
> That should fix the bug while also ensuring individual labels can still be passed via the pipeline into other functions.

I thought of this approach too, but I thought it was a mistake that would add too much complexity if we try and treat this as a Label object when that was not the original intention / format of the original Github API, but I can go this route if you prefer.

**Comment by joseartrivera** [comment]
> Thanks for the report. I was able to repro your issue by calling
> 
> ```powershell
> Get-GitHubEvent -OwnerName microsoft -RepositoryName PowerShellForGitHub -Issue 306
> ```
> 
> What you supplied would indeed fix the specific issue you're running into, but I don't know if it's actually the _right_ fix.
> 
> I think the more complete fix would be to create a new `Type` called `GitHub.LabelSummary` for the reduce content, similar to what we do for [`GitHub.Team`/`GitHub.TeamSummary`](https://github.com/microsoft/PowerShellForGitHub/blob/d4997057f8b1234ea1aabeb4fb6742148d3afaaf/GitHubTeams.ps1#L5-L6), and then update `Add-GitHubLabelAdditionalProperties` to optionally take in the `OwnerName` and `RepositoryName`. If supplied, and `$item.url` is not defined, then those should be used for forming the `$repositoryUrl`. Then you'd update [`Add-GitHubEventAdditionalProperties`](https://github.com/microsoft/PowerShellForGitHub/blob/d4997057f8b1234ea1aabeb4fb6742148d3afaaf/GitHubEvents.ps1#L238-L246) to pass in the `ReposistoryUrl` and `OwnerName` along with the new `TypeName`.
> 
> That should fix the bug while also ensuring individual labels can still be passed via the pipeline into other functions.

I updated the code, there is no need to update the params to GitHubEventAdditionalProperties as the repository url is already available from the base event object.

**Review by HowardWolosky — CHANGES_REQUESTED** [review]
Thanks for the update.  Some minor additional updates being requested here.

File: GitHubLabels.ps1 [hunk 1]
```diff
@@ -1330,6 +1330,10 @@ filter Add-GitHubLabelAdditionalProperties
     .PARAMETER TypeName
         The type that should be assigned to the object.
 
+    .PARAMETER RepositoryUrl
+        Optionally supplied if the Label object doesn't have this value already
+        (as is the case for GitHub.LabelSummary).
```
Comment (HowardWolosky) [inline GitHubLabels.ps1:1335]: These should be listed in the same order as the parameters.

```suggestion
    .PARAMETER RepositoryUrl
        Optionally supplied if the Label object doesn't have this value already
        (as is the case for GitHub.LabelSummary).

    .PARAMETER TypeName
        The type that should be assigned to the object.
```

File: GitHubLabels.ps1 [hunk 2]
```diff
@@ -1356,8 +1362,16 @@ filter Add-GitHubLabelAdditionalProperties
...
+            if ([System.String]::IsNullOrEmpty($item.url) -and $PSBoundParameters.ContainsKey('RepositoryUrl'))
+            {
+                $repositoryUrl = $RepositoryUrl
+            }
+            else
+            {
+                $elements = Split-GitHubUri -Uri $item.url
+                $repositoryUrl = Join-GitHubUri @elements
+            }
+
```
Comment (HowardWolosky) [inline GitHubLabels.ps1:1374]: Simplifying

```suggestion
            if (-not [System.String]::IsNullOrEmpty($item.url))
            {
                $elements = Split-GitHubUri -Uri $item.url
                $RepositoryUrl = Join-GitHubUri @elements
            }

```

File: GitHubLabels.ps1 [hunk 3]
```diff
@@ -1356,8 +1362,16 @@ filter Add-GitHubLabelAdditionalProperties
...
             Add-Member -InputObject $item -Name 'RepositoryUrl' -Value $repositoryUrl -MemberType NoteProperty -Force
```
Comment (HowardWolosky) [inline GitHubLabels.ps1:1375]:
```suggestion
            Add-Member -InputObject $item -Name 'RepositoryUrl' -Value $RepositoryUrl -MemberType NoteProperty -Force
```

**Review by HowardWolosky — APPROVED** [review]
Thanks for the update.
There are some additional changes to make, but I've already supplied the fixes as suggestions and can apply them myself once this review is submitted.

**Review by HowardWolosky — APPROVED** [review]

**Pull request merged**